File: src/ru/greeneyes/project/pomidoro/PomodoroAction.java
package ru.greeneyes.project.pomidoro;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.application.ApplicationManager;

/**
 * User: dima
 * Date: May 30, 2010
 */
public class PomodoroAction extends AnAction {
	@Override
	public void actionPerformed(AnActionEvent event) {
		PomodoroComponent pomodoroComponent = ApplicationManager.getApplication().getComponent(PomodoroComponent.class);
		pomodoroComponent.getModel().switchToNextState();
	}
}

File: src/pomodoro/PomodoroAction.kt
package pomodoro

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.components.service
import com.intellij.openapi.project.DumbAware
import pomodoro.model.time.Time

class PomodoroAction : AnAction(), DumbAware {
    override fun actionPerformed(event: AnActionEvent) {
        service<PomodoroComponent>().model.onUserSwitchToNextState(Time.now())
    }
}
